src/ios/CDVCookieMaster.m

```diff
@@ -71,11 +71,22 @@ - (void)setCookieValue:(CDVInvokedUrlCommand*)command
 
 - (void)clearCookies:(CDVInvokedUrlCommand*)command
 {
-    NSHTTPCookie *cookie;
-    NSHTTPCookieStorage *storage = [NSHTTPCookieStorage sharedHTTPCookieStorage];
-    for (cookie in [storage cookies]) {
-        [storage deleteCookie:cookie];
+    if (@available(iOS 11.0, *)) {
+        NSSet *websiteDataTypes = [NSSet setWithArray:@[WKWebsiteDataTypeCookies]];
+        NSDate *dateFrom = [NSDate dateWithTimeIntervalSince1970:0];
+        [[WKWebsiteDataStore defaultDataStore] removeDataOfTypes:websiteDataTypes
+                                                modifiedSince:dateFrom
+                                                completionHandler:^{
+                                                    NSLog(@"Cookies Cleared");
+                                                }];
+    } else {
+        NSHTTPCookie *cookie;
+           NSHTTPCookieStorage *storage = [NSHTTPCookieStorage sharedHTTPCookieStorage];
+           for (cookie in [storage cookies]) {
+               [storage deleteCookie:cookie];
+           }
     }
+
     [[NSUserDefaults standardUserDefaults] synchronize];
 
     CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
```

Comment:
Similar to how WKWebView support was added for `setCookieValue`, we should be wrapping the iOS 11 check in a check to see if the app is running in WKWebView. This could otherwise create issues for apps running in UIWebView. I know that Apple will no longer allow updates of apps using UIWebView beginning in December 2020, but for now we should continue to support it. 🙂 Here's the check I'm referring to:
https://github.com/Cartegraph/cordova-cookie-master/commit/9f7fb02ecbd4373b23b527e1cb19acee04c20741#diff-4a1ce2d830a973505b873c255085e7b6R61

Below are my suggested changes for this PR. Besides the addition of the WKWebView check and handling the failure case, I moved the `synchronize` call into the UIWebView branch of logic because I don't think it's necessary going forward with WKWebView. See here: https://developer.apple.com/documentation/foundation/nsuserdefaults/1414005-synchronize?language=objc . I also adjusted some code indentation.

Thanks a lot for contributing the PR! Let me know your thoughts. 🙂

```suggestion
    CDVPluginResult* pluginResult = nil;
    
    if ([self.webView isKindOfClass:[WKWebView class]]) {
        if (@available(iOS 11.0, *)) {
            NSSet *websiteDataTypes = [NSSet setWithArray:@[WKWebsiteDataTypeCookies]];
            NSDate *dateFrom = [NSDate dateWithTimeIntervalSince1970:0];
            [[WKWebsiteDataStore defaultDataStore] removeDataOfTypes:websiteDataTypes
                                                    modifiedSince:dateFrom
                                                    completionHandler:^{
                                                        NSLog(@"Cookies Cleared");
                                                    }];
        } else {
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"WKWebView requires iOS 11+ in order to clear cookies"];
            [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];

            return;
        }
    } else {
        NSHTTPCookie *cookie;
        NSHTTPCookieStorage *storage = [NSHTTPCookieStorage sharedHTTPCookieStorage];
        for (cookie in [storage cookies]) {
            [storage deleteCookie:cookie];
        }
           
        [[NSUserDefaults standardUserDefaults] synchronize];
    }
    
    pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
```